docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,531 @@
+
+If a build fails, click on its `Details` link for more information.
+
+For example, clicking on the `Details` link for the `Travis CI build` job takes
+us to the Travis CI build page.
+
+![submit_pr_travis_ci_build_page_1.png](./imgs/submit_pr_travis_ci_build_page_1.png)
+
+Here you can investigate the status of the job, its sub-jobs, or restart them.
+
+#### 5. PR Review Process
+
+The review process is iterative, so it is normal that `multiple commits` may be
+required until the pull request is finally accepted.
+
+{% include tip.html content="After a pull request is created, any additional commits that you push to your remote feature branch automatically get added to the pull request. There is no need to close the initial pull request and create a new one." %}
+
+Reactively [rebase](https://help.github.com/articles/about-git-rebase/) your
+forked repository against the upstream master branch to keep them in sync; if
+needed.
+
+#### Before your PR is merged by a maintainer
+
+Before the PR is merged by a
+maintainer,[squash your commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)
+into one commit or a few commits; if needed. This step helps to keep the
+repository's commit history clean and concise.
+
+### Beginner Instructions
+
+Here is a complete tutorial on how to submit a pull request (PR) for LoopBack
+v4.
+
+#### 1. Fork the loopback-next repository
+
+In your browser, navigate to : <https://github.com/strongloop/loopback-next> .
+
+Create your own fork of the repository, by pressing the `Fork` link on the
+right-hand side .
+
+![submit_pr_fork_main_repo.png](./imgs/submit_pr_fork_main_repo.png)
+
+When the forking process is complete, the repository will show up as
+`{ your user id }/loopback-next`. In my case, it is `emonddr/loopback-next` .
+
+![submit_pr_my_forked_repo.png](./imgs/submit_pr_my_forked_repo.png)
+
+#### 2. Create the feature branch
+
+Notice your repo has a `master` branch already created (refer to bottom left
+corner of the picture above). It is commonplace to have this branch represent
+the latest, clean version of this repository's content. For the purposes of your
+PR, let's create a `feature` branch with a name indicative of your changes. In
+my case, it is `emonddr-doc-changes`.
+
+Click on the `Clone or download` button
+
+![submit_pr_create_feature_branch_1.png](./imgs/submit_pr_create_feature_branch_1.png)
+
+This brings up tiny dialog with different choices : Clone with SSH, Use HTTPS,
+Open in Desktop, or Download Zip.
+
+In my case, I will leave it as `Clone with SSH`, and click on the
+`copy to clipboard` icon on the right side of the repository url (we are going
+to paste this value in a terminal window soon).
+
+![submit_pr_create_feature_branch_2.png](./imgs/submit_pr_create_feature_branch_2.png)
+
+Open a terminal window, and navigate to the directory where you want to clone
+the repository. In my case, this is `/Users/dremond/git` .
```

Comment:
nitpick: Using `git` as the directory is confusing. It would be better to use something like `/Users/dremond/Projects`.